packages/machine/src/protocol/install-virtual-app.ts

```diff
@@ -150,40 +172,25 @@ function createTarget(
       token: AddressZero
     },
     true, // sets it to be a virtual app
-    0, // app seq no: virtual app instances do not have appSeqNo
+    sc.numInstalledApps, // app seq no
     0, // root nonce value: virtual app instances do not have rootNonceValue
     initialState,
     0, // app nonce
     defaultTimeout
   );
-}
-
-function addTarget(
-  context: Context,
-  initiatingXpub: string,
-  respondingXpub: string,
-  intermediaryXpub: string,
-  targetAppInstance: AppInstance
-) {
-  const key = virtualChannelKey(
-    [initiatingXpub, respondingXpub],
-    intermediaryXpub
-  );
 
-  const sc = (
-    context.stateChannelsMap.get(key) ||
-    StateChannel.createEmptyChannel(key, [initiatingXpub, respondingXpub])
-  ).addVirtualAppInstance(targetAppInstance);
+  const newSc = sc.addVirtualAppInstance(target);
```

Comment:
Can you add a check inside `addVirtualAppInstance` to verify the correct `appSeqNo` was set?